nibabel/nicom/dicomwrappers.py

```diff
@@ -461,10 +461,23 @@ def __init__(self, dcm_data):
         Wrapper.__init__(self, dcm_data)
         self.dcm_data = dcm_data
         self.frames = dcm_data.get('PerFrameFunctionalGroupsSequence')
+        self._nframes = self.get('NumberOfFrames')
         try:
             self.frames[0]
         except TypeError:
             raise WrapperError("PerFrameFunctionalGroupsSequence is empty.")
+        try:
+            # DWI image where derived isotropic, ADC or trace volume
+            # was appended to the series
+            if self.frames[0].get([0x18, 0x9117], None):
+                self.frames = Sequence(
+                    frame for frame in self.frames if
+                    frame.get([0x18, 0x9117])[0].get([0x18, 0x9075]).value
+                    != 'ISOTROPIC'
+                    )
+                self._nframes = len(self.frames)
+        except AttributeError:
+            pass
```

Comment:
This is a pretty broad `try`/`except` block. Can the expected failures be narrowed down at all?

Is there a possibility of an `IndexError` with `frame.get([0x18, 0x9117])[0]`? If so, is that an indicator of corrupt data where we should raise a `WrapperError`?